lib/order.rb

```diff
@@ -0,0 +1,77 @@
+require_relative 'menu'
+require_relative 'text_message'
+
+class Order
+
+  attr_reader :basket, :menu
+
+  def initialize(menu = Menu.new)
+    @basket = Hash.new(0)
+    @menu = menu
+  end
+
+  def add_dish(item)
+    @menu.check_availability(item)
+    @basket[item] += @menu.dishes[item]
+  end
+
+  def remove_dish(item)
+    check_basket(item)
+    @basket[item] -= @menu.dishes[item]
+  end
+
+  def view_basket
+    check_empty_basket
```

Comment:
Minor one here - why raise an error, rather than doing a fallback to printing an empty basket? Usually, raising exceptions is done to prevent an error/illegal situation from happening, but we could think of having an empty basket an "ok" case, it would just print an empty basket with a total of 0. But that's probably a subjective discussion here :D

Reply:
You're totally right actually! I think I just like this raise error method too much, so I'm putting it everywhere :D The view_basket method should just print an empty basket as you said. I will amend it. Thanks!